versions/development/SPEC.md

```diff
@@ -302,7 +302,37 @@ For more details on the postfix quantifiers, see the section on [Optional Parame
 
 For more information on type and how they are used to construct commands and define outputs of tasks, see the [Data Types & Serialization](#data-types--serialization) section.
 
-#### Custom  Types
+#### Numeric Types
+
+`Int` and `Float` are the numeric types.
+Int can be used to hold any integer, there is no size limit.
+Float is used to hold a finite 64-bit IEEE-754 floating point number.
+Note that WDL itself is not intended to be used for heavy numeric computation.
+There are no number formatting utilities, many mathematical functions are absent,
+and engine behaviour at extreme values is not defined and should not be relied upon.
+
+The mathematical operators are defined only for identical types.
+In expressions using an operator with mismatched numeric types,
+the `Int` will be cast to `Float` and the result will be `Float`.
+Note that this will cause loss of precision. The `Float` can be
+converted to `Int` with the `ceil`, `round` or `floor` functions if needed.
+
+In expressions involving a numeric type and a String, the numeric type
+will be converted to a string. This is straigthforward for `Int`.
+For `Float`, behaviour will be similar to the 'g' conversion specifier
+defined by the C language.
+If you have stricter requirements for `Float` formatting, you should use `String`
+and call out to your language or tool of choice for numerical manipulation.
+
+#### String Types
+
+`String` and `File` are the string types.
+`File` type can be used in place of `String` in expressions.
```

Comment:
So that this : 
```
File input ...
File output = "${input}.md5"
``` 
would be valid, but not this : 
```
File output = input + ".md5"
```

I think it makes great sense to me. Is the wordage then something like; 

```String interpolation (${}) expressions must evaluate to a non-collection type, ints floats strings bools formatted as normal blah, File is substituted to the URI depending on the engine.

Only boolean operators are defined for File (Maybe not even them?). 

Files can only be declared (=) from a string or a file. If string, it must make sense to the engine (URI or path)
```